sdk/export/metric/metric.go

```diff
@@ -99,35 +106,39 @@ type AggregationSelector interface {
 // MinMaxSumCount aggregator to a Counter instrument.
 type Aggregator interface {
 	// Update receives a new measured value and incorporates it
-	// into the aggregation.  Update() calls may arrive
-	// concurrently as the SDK does not provide synchronization.
+	// into the aggregation.  Update() calls may be called
+	// concurrently.
 	//
 	// Descriptor.NumberKind() should be consulted to determine
 	// whether the provided number is an int64 or float64.
 	//
 	// The Context argument comes from user-level code and could be
-	// inspected for distributed or span context.
+	// inspected for a `correlation.Map` or `trace.SpanContext`.
 	Update(context.Context, metric.Number, *metric.Descriptor) error
 
-	// Checkpoint is called during collection to finish one period
-	// of aggregation by atomically saving the current value.
-	// Checkpoint() is called concurrently with Update().
-	// Checkpoint should reset the current state to the empty
-	// state, in order to begin computing a new delta for the next
-	// collection period.
+	// SynchronizedCopy is called during collection to finish one period
+	// of aggregation by atomically saving the current state into
+	// the argument Aggregator.
 	//
-	// After the checkpoint is taken, the current value may be
-	// accessed using by converting to one a suitable interface
-	// types in the `aggregator` sub-package.
+	// SynchronizedCopy() is called concurrently with Update().  These
+	// two methods must be synchronized with respect to each
+	// other, for correctness.
+	//
+	// After saving a synchronized copy, the Aggregator can be converted
+	// into one or more of the interfaces in the `aggregation` sub-package,
```

Comment:
The aggregators are not interfaces anymore, is there an end goal in mind here to switch back?

Reply:
This comment confused me--somewhat. I think of the interfaces as having stayed the same, only the `Checkpoint()` method changed / renamed. Whereas before we converted a two-state `export.Aggregator` into one of the `sdk/export/metric/aggregation` interfaces (e.g., Sum, LastValue, ...), getting the checkpointed half of the object, now we convert the one-state `export.Aggregator` into another one of the `sdk/export/metric/aggregation` interfaces. 

For me a takeaway here, is that there is a lot of complexity. :(